### PR TITLE
feat: extend s3 compatible config

### PR DESCRIPTION
**Pull request by warjiang**
1.  if user set the s3 endpoint,`goreleaser` will set `s3ForcePathStyle=true` in the query :
image
this pr want to allow user make choice whether to set `s3ForcePathStyle=true`

2.  allow user set s3 upload policy, wants goreleaser support s3 acl policy 

In short, user will use goreleaser  s3 provider like this:
image

### CODE REVIEW

**Comment by caarlos0** [comment]
those options would apply to all the files... I think, specially the ACLs, people will want to customize per file


check #3069

**Comment by warjiang** [comment]
> those options would apply to all the files... I think, specially the ACLs, people will want to customize per file
> 
> check #3069

As you mentioned, the ACL policy will apply to all ids in the config file。But there is a trick way can specify differen acl for artifact.  If the original config like this:
```
blobs:
  - provider: s3
    endpoint: {{s3-endpoint}}
    region: {{s3-region}}
    disableSSL: true
    bucket:{{s3-bucket}}
    s3ForcePathStyle: false
    ids:
      - a
      - b
    folder: {{folder}}
```

if we want `a` be `public-read` acl, and `b` be `private` acl, we can do it like this:
```
blobs:
 - provider: s3
    endpoint: {{s3-endpoint}}
    region: {{s3-region}}
    disableSSL: true
    bucket:{{s3-bucket}}
    s3ForcePathStyle: false
    acl: public-read
    ids:
      - a
    folder: {{folder}
 - provider: s3
    endpoint: {{s3-endpoint}}
    region: {{s3-region}}
    disableSSL: true
    bucket:{{s3-bucket}}
    s3ForcePathStyle: false
    acl: public-read
    ids:
      - b
    folder: {{folder}}
```

**Comment by warjiang** [comment]
By the way `s3ForcePathStyle` cannot specify for each file,`s3ForcePathStyle` should come with s3 endpoint,  more info can visit https://docs.aws.amazon.com/AmazonS3/latest/userguide/VirtualHosting.html

image

File: internal/pipe/blob/upload.go [hunk 1]
```diff
@@ -230,15 +239,27 @@ func (u *productionUploader) Open(ctx *context.Context, bucket string) error {
...
 
 func (u *productionUploader) Upload(ctx *context.Context, filepath string, data []byte) error {
 	log.WithField("path", filepath).Info("uploading")
-
 	opts := &blob.WriterOptions{
 		ContentDisposition: "attachment; filename=" + path.Base(filepath),
 	}
+	if u.conf.ACL != "" {
+		opts.BeforeWrite = func(asFunc func(interface{}) bool) error {
+			req := &s3manager.UploadInput{}
```
Comment (caarlos0) [inline internal/pipe/blob/upload.go:253]: what if the bucket is not on s3? wouldn't this fail?

**Pull request closed without merging**